plugins/home/src/api/VisitsStorageApi.ts

```diff
@@ -117,23 +117,33 @@ export class VisitsStorageApi implements VisitsApi {
   }
 
   private async persistAll(visits: Array<Visit>) {
-    const { userEntityRef } = await this.identityApi.getBackstageIdentity();
-    const storageKey = `${this.storageKeyPrefix}:${userEntityRef}`;
-
+    const storageKey = await this.getStorageKey();
     return this.storageApi.set<Array<Visit>>(storageKey, visits);
   }
 
   private async retrieveAll(): Promise<Array<Visit>> {
+    const storageKey = await this.getStorageKey();
+    // Handles for case when snapshot is and is not referenced per storaged type used
+    const snapshot = this.storageApi.snapshot<Array<Visit>>(storageKey);
+    if (snapshot?.presence !== 'unknown') {
+      return snapshot?.value ?? [];
+    }
+
+    return new Promise((resolve, reject) => {
+      this.storageApi.observe$<Visit[]>(storageKey).subscribe({
```

Comment:
```suggestion
      const subscription = this.storageApi.observe$<Visit[]>(storageKey).subscribe({
```

and then call `subscription.unsubcribe()` both in `next` and `error` below to ensure that we clean up resources

Reply:
@freben just pushed the change up. Tested locally as well to confirm it still functions